### PR TITLE
Write session progress to the client side cookie for anonymous users

### PR DESCRIPTION
**Pull request by mehalshah**

### CODE REVIEW

**Comment by philbogle** [comment]
Reading

File: dashboard/test/controllers/activities_controller_test.rb [hunk 1]
```diff
@@ -715,42 +624,6 @@ def expect_controller_logs_milestone_regexp(regexp)
...
-
-    # do all the logging
-    @controller.expects :log_milestone
-    @controller.expects :slog
-
-    @controller.expects(:trophy_check).never # no trophy if not logged in
-
-    expect_s3_upload
-
-    assert_creates(LevelSource, LevelSourceImage) do
```
Comment (philbogle) [inline dashboard/test/controllers/activities_controller_test.rb:733]: Many of the assertions in this test are still relevant; we want to keep this test and only get rid of the assertions about client_state (since that state is now updated on the client.)

File: dashboard/test/controllers/activities_controller_test.rb [hunk 2]
```diff
@@ -628,69 +600,6 @@ def expect_controller_logs_milestone_regexp(regexp)
...
-    @controller.expects(:trophy_check).never # no trophy if not logged in
-
-    assert_creates(LevelSource) do
-      assert_does_not_create(Activity, UserLevel) do
-        post :milestone, @milestone_params.merge(user_id: 0)
-      end
-    end
-
-    # record activity in session
-    assert_equal 50, client_state.level_progress(@script_level_prev.level_id)
```
Comment (philbogle) [inline dashboard/test/controllers/activities_controller_test.rb:680]: Likewise we should keep this test and only get rid of the assertions about server side updates of client state.

File: dashboard/test/controllers/activities_controller_test.rb [hunk 3]
```diff
@@ -886,9 +759,7 @@ def new_level
     assert_equal true, new_level
     assert_equal false, new_level
 
-    sign_out @user
```
Comment (philbogle) [inline dashboard/test/controllers/activities_controller_test.rb:889]: Even for anonymous users we should try to POST to milestone and get back the same new level response (I think), please let me know if that's not the case.

**Comment by philbogle** [comment]
The code looks good; there were some dropped tests that I think we still want.

File: dashboard/test/controllers/activities_controller_test.rb (hunk 3 above)
Comment (mehalshah) [inline dashboard/test/controllers/activities_controller_test.rb:889]: So we can do the first assert, but not the second one.

File: aws/build_and_mail_log [hunk 4]
```diff
@@ -35,7 +35,7 @@ def main()
     commit_url = "https://github.com/code-dot-org/code-dot-org/commit/#{RakeUtils.git_revision}"
     # Notify the HipChat channels about what happened.
     projects = ARGV.join(' ')
-    time_message = " &#x1f550; #{format_duration(Time.now.to_i - start_time)}"
+    time_message = " 🕐 #{format_duration(Time.now.to_i - start_time)}"
```
Comment (philbogle) [inline aws/build_and_mail_log:38]: Did you really want this change?  It seems unrelated to other change and not sure we need it.  Might also lead to issues if some tool gets confused about encoding of this file.

File: dashboard/test/controllers/activities_controller_test.rb [hunk 5]
```diff
@@ -218,34 +218,6 @@ def expect_controller_logs_milestone_regexp(regexp)
     assert_equal 1000, Activity.last.lines
   end
 
-  test "anonymous milestone does not allow unreasonably high lines of code" do
```
Comment (philbogle) [inline dashboard/test/controllers/activities_controller_test.rb:221]: Does this functionality need to be reimplemented in Javascript?

File: dashboard/test/controllers/activities_controller_test.rb [hunk 6]
```diff
@@ -643,16 +615,10 @@ def expect_controller_logs_milestone_regexp(regexp)
...
-    assert_equal 100, client_state.level_progress(@script_level.level_id)
-
-    # record the total lines of code in session
-    assert_equal 20, client_state.lines
-
     assert_response :success
 
     expected_response = build_expected_response(
-        total_lines: 20,
+        total_lines: 0,
```
Comment (philbogle) [inline dashboard/test/controllers/activities_controller_test.rb:621]: Is it the case that the response includes total_lines but it is no longer meaningful since it is computed client side? 
Might be better to just eliminate that from the response and make sure there is no client code referencing it, since if there were such client code, that code would be referencing bogus values and buggy.

File: aws/build_and_mail_log (hunk 4 above)
Comment (mehalshah) [inline aws/build_and_mail_log:38]: Not sure how this got included. I'm going to back it out.

File: dashboard/test/controllers/activities_controller_test.rb (hunk 5 above)
Comment (mehalshah) [inline dashboard/test/controllers/activities_controller_test.rb:221]: I'll add the functionality as well as something in client_side_tests

File: dashboard/test/controllers/activities_controller_test.rb (hunk 6 above)
Comment (mehalshah) [inline dashboard/test/controllers/activities_controller_test.rb:621]: Caveat-- total_lines from the server might be meaningful in the signed in case but not the signed out case, so consider that possibility!

File: aws/build_and_mail_log (hunk 4 above)
Comment (mehalshah) [inline aws/build_and_mail_log:38]: This was part of a change that Josh made - think it was errantly included in this diff

File: dashboard/test/controllers/activities_controller_test.rb (hunk 6 above)
Comment (mehalshah) [inline dashboard/test/controllers/activities_controller_test.rb:621]: I don't think total_lines is used on the server end for anonymous users. There's a call to track_progress_for_user (signed in user) and track_progress_in_session (signed out user), the latter is used and does not do anything with total_lines
